### ISSUE TITLE
Bash Completion

### ISSUE THREAD

**Issue by Cronos87** [timvisee/ffsend#28]
Hello,

Will you add a bash completion script to help using it?

I can try to help you with this if you want.

Thank you for your answer.

**Comment by timvisee** [comment]
I've recently added shell completion files for some common shells, which 
can be found in 
[`./contrib/completions`](https://github.com/timvisee/ffsend/tree/master/contrib/completions), 
the `bash` completion script is named 
[`ffsend.bash`](https://github.com/timvisee/ffsend/blob/master/contrib/completions/ffsend.bash). 


I haven't thoroughly tested these completion scripts though, thus it 
would be awesome if you could give it a try.

I've added the `ffsend generate completions` command in 
[`v0.2.37`](https://github.com/timvisee/ffsend/releases/tag/v0.2.37) to 
allow you to generate these files yourself. You might find it useful.

None of these completions have been added to any of the packages yet, 
though. Manual installation of these scripts is currently required.

**Comment by timvisee** [comment]
I've recently added shell completion files for some common shells, which 
can be found in [`./contrib/completions`](https://github.com/timvisee/ffsend/tree/master/contrib/completions), the `bash` completion script is named [`ffsend.bash`](https://github.com/timvisee/ffsend/blob/master/contrib/completions/ffsend.bash). 

I haven't thoroughly tested these completion scripts though, thus it would be awesome if you could give it a try.

I've added the `ffsend generate completions` command in [`v0.2.37`](https://github.com/timvisee/ffsend/releases/tag/v0.2.37) to allow you to generate these files yourself. You might find it useful.

None of these completions have been added to any of the packages yet, though. Manual installation of these scripts is currently required.

**Comment by Cronos87** [comment]
Oh that's awesome!

I will give it a try! Thank you.

By the way, maybe you can ask the author of the homebrew script #21 to add the completion script as reference in the homebrew installation script. It's possible to do this. It will be awesome to have it.

**Comment by timvisee** [comment]
Yes, that's a good idea.

It might be a good idea to wait with completions for `homebrew` though, 
until the first package definition is merged.

**Comment by timvisee** [comment]
Yes, that's a good idea.

It might be a good idea to wait with completions for `homebrew` though, until the first package definition is merged.

**Comment by matevzmihalic** [comment]
```
    bash_completion.install "contrib/completions/ffsend.bash"
    fish_completion.install "contrib/completions/ffsend.fish"
    zsh_completion.install "contrib/completions/_ffsend"
```

You can add this to install block of homebrew formula, before you do the next update, to install completions through homebrew.

**Comment by Cronos87** [comment]
Hello!

Will you add this in the near future?

Thank you 🙂

**Comment by timvisee** [comment]
Sorry for not getting back on this.

I've opened a PR for the Homebrew package to install these completion files.  
See: https://github.com/Homebrew/homebrew-core/pull/43546

Assuming this PR is accepted, the completion files should be installed with the next update through Homebrew. I've just updated the completion files as well, to make sure the match the latest CLI interface changes.

**Comment by timvisee** [comment]
I've just released `v0.2.51`, which should include (and automatically install) shell completions for common shells with the Homebrew and AUR packages.

This release, it's binaries and packages should be available as soon as this release pipeline succeeds:  
https://gitlab.com/timvisee/ffsend/pipelines/79258954

I'll close this issue for now, assuming your original issue is fixed with this update. I'll attempt to add completions in other packages as well in the future.
